libsolidity/ast/Types.cpp

```diff
@@ -2555,7 +2555,7 @@ bool UserDefinedValueType::operator==(Type const& _other) const
 
 string UserDefinedValueType::toString(bool /* _short */) const
 {
-	return "user defined type " + definition().name();
+	return "user defined type " + *definition().annotation().canonicalName;
```

Comment:
This is not fully related to this PR, but why does this have this free form text, when the other cases use strings which look more like code, such as `enum X`, `tuple(int256,int256)`, etc.

Reply:
What would be your suggestion? Just `type MyInt`?

Reply:
I wonder if we even need a special keyword, but `type MyInt` does sound good enough.

Mostly unsure about what is the use case for this, in the AST the `typeString` for mappings looks like valid Solidity syntax. If it is kind of supposed to be valid syntax, then this would break it.

Reply:
This is currently used in error messages such as "Type user defined type L.T is not implicitly convertible to expected type user defined type C.T."

If we change to just `type MyInt`, the above error message would look a bit weird:

"Type type L.T is not implicitly convertible to expected type type C.T."

maybe `userdefined MyInt` is better?

Reply:
Yeah, `userdefined` is perhaps better. Why do we need to mark it specially though and not just by its name? Built in types can't be overridden.

Reply:
> Why do we need to mark it specially though and not just by its name

There were some errors where 
```solidity
contract C {
type MyInt is int;
}
contract D {
type MyInt is int;
}
contract D {
// The error message for this also need to mention the scope
C.MyInt x = D.MyInt.wrap(0);
}
```

Reply:
I think it's nice to know if it's a struct, a contract or a user-defined type.

Reply:
Didn't we say structs **are** user-defined types, so "user-defined type" is a bad designation for the user-defined-value-type things :-)?